SIS_import/sis_upload.rb

```diff
@@ -208,13 +213,19 @@ def upload_to_canvas(fileName, output_file_base_name)
 			end
 			## hashmap ["message"=>"error_message"
 			$logger.warn "upload error: #{upload_error}"
-
 			break
-		else
-			job_progress=parsed_result["progress"]
-			$logger.info "Canvas upload job processed #{job_progress}"
 		end
-	end until job_progress == 100
+	end until (!workflow_state.eql?("created") && !workflow_state.eql?("importing"))
+	# "workflow_status" is a better indicator of the upload progress, instead of the "progress" field
+	# we have seen example of "progress=100" while "workflow_statue=importing"
+	# Possible value for "workflow_state":
+	# - 'created': The SIS import has been created.
+	# - 'importing': The SIS import is currently processing.
+	# - 'imported': The SIS import has completed successfully.
+	# - 'imported_with_messages': The SIS import completed with errors or warnings.
+	# - 'failed_with_messages': The SIS import failed with errors, while the import completed partially (or mostly)
+	# - 'failed': The SIS import failed and the upload process did not complete at all
+	# either 'failed_with_messages' or 'failed' would be caused usually by a corrupt SIS file or typos in the SIS.
 
 	if (!upload_error)
```

Comment:
Does this mean "*Also* generate upload_error when the workflow_status....."?

- Dave

On Mon, Sep 14, 2015 at 8:58 AM, Zhen Qian <notifications@github.com> wrote:

> In SIS_import/sis_upload.rb
> <https://github.com/tl-its-umich-edu/canvas/pull/72#discussion_r39389869>:
>
> >  		end
> > -	end until job_progress == 100
> > +	end until (!workflow_state.eql?("created") && !workflow_state.eql?("importing"))
> > +	# "workflow_status" is a better indicator of the upload progress, instead of the "progress" field
> > +	# we have seen example of "progress=100" while "workflow_statue=importing"
> > +	# Possible value for "workflow_state":
> > +	# - 'created': The SIS import has been created.
> > +	# - 'importing': The SIS import is currently processing.
> > +	# - 'imported': The SIS import has completed successfully.
> > +	# - 'imported_with_messages': The SIS import completed with errors or warnings.
> > +	# - 'failed_with_messages': The SIS import failed with errors, while the import completed partially (or mostly)
> > +	# - 'failed': The SIS import failed and the upload process did not complete at all
> > +	# either 'failed_with_messages' or 'failed' would be caused usually by a corrupt SIS file or typos in the SIS.
> >
> >  	if (!upload_error)
>
> Done. Generate upload_error when the workflow_status is 'failed' and no
> error message is returned from API call.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tl-its-umich-edu/canvas/pull/72/files#r39389869>.
>